server/source/world/TerrainGenerator.cpp

```diff
@@ -115,16 +117,16 @@ void TerrainGenerator::fastNoiseGeneration(ServerChunk &chunk) const {
 				}
 				else {
 					if (z + chunk.z() * CHUNK_HEIGHT >= h - 1 && z + chunk.z() * CHUNK_HEIGHT > SEALEVEL - 1)
-						chunk.setBlockRaw(x, y, z, m_grassBlockID);
+						chunk.setBlockRaw(x, y, z, biome.getTopBlock());
 					else if (z + chunk.z() * CHUNK_HEIGHT <= SEALEVEL - 1 && h < SEALEVEL && z + chunk.z() * CHUNK_HEIGHT > h - 3)
-						chunk.setBlockRaw(x, y, z, m_sandBlockID);
+						chunk.setBlockRaw(x, y, z, biome.getBeachBlock());
 					else if (z + chunk.z() * CHUNK_HEIGHT > h - 3)
-						chunk.setBlockRaw(x, y, z, m_dirtBlockID);
+						chunk.setBlockRaw(x, y, z, biome.getGroundBlock());
 					else
-						chunk.setBlockRaw(x, y, z, m_stoneBlockID);
+						chunk.setBlockRaw(x, y, z, biome.getDeepBlock());
 
 					if ((rand() % 4096) == 0)
-						oreFloodFill(chunk, x, y, z, m_stoneBlockID, m_ironOreBlockID, 2);
+						oreFloodFill(chunk, x, y, z, biome.getDeepBlock(), m_ironOreBlockID, 2);
```

Comment:
We can probably add a new optional table named `ores` in the Lua biome definition, like so:
```lua
ores = {
    { id="default:iron_ore", rarity="0.2" },
    { id="default:coal_ore", rarity="0.1" }
}
```

Reply:
Note: `rarity` can be unused by the biome generator, if you prefer, you can even simplify it to start easier:
```lua
ore = "default:iron_ore"
```